apps/wallpaper/js/share.js

```diff
@@ -26,11 +26,14 @@ window.onload = function() {
   var scale;
   var previewImage = document.getElementById('previewImage');
 
-  // How big (in device pixels) is the screen?
-  var screenWidth = Math.ceil(screen.width * window.devicePixelRatio);
-  var screenHeight = Math.ceil(screen.height * window.devicePixelRatio);
+  var screenWidth;
+  var screenHeight;
 
   function startShare(request) {
+    // How big (in device pixels) is the screen?
+    screenWidth = Math.ceil(screen.width * window.devicePixelRatio);
+    screenHeight = Math.ceil(screen.height * window.devicePixelRatio);
+
```

Comment:
It makes me nervous to just move these assignments here, given the async nature of resize events.  Maybe it would be better to leave the initial assignments where they were, and also register a resize handler to detect when they change.

Or, since this is for wallpaper, we could always just take the longer dimension as the height and the shorter as the width:

```
screenWidth = Math.ceil(Math.min(screen.width, screen.height) * window.devicePixelRatio);
screenHeight = Math.ceil(Math.max(screen.width, screen.height) * window.devicePixelRatio);
```